augur/tasks/util/collection_util.py

```diff
@@ -36,6 +36,168 @@ class CollectionState(Enum):
+            AND augur_operations.collection_status.ml_data_last_collected IS NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+        """
+    else:
+        condition_concat_string = f"""
+            ml_status='Success' AND ml_status!='{str(CollectionState.ERROR.value)}'
+            AND augur_operations.collection_status.ml_data_last_collected IS NOT NULL
+            AND ml_status!='{str(CollectionState.COLLECTING.value)}'
+            AND ml_data_last_collected <= NOW() - INTERVAL '{days_until_collect_again} DAYS'
+        """
+    
+    return condition_concat_string
+
+class CollectionRequest:
+    def __init__(self,name,phases,max_repo = 10,new_status = CollectionState.PENDING.value,additional_conditions = None,days_until_collect_again = 1):
+        self.name = name
+        self.phases = phases
+        self.max_repo = max_repo
+        self.days_until_collect_again = days_until_collect_again
+        self.additional_conditions = None
+        self.new_status = new_status
+        self.repo_list = []
+
+        self.status_column = f"{name}_status"
+
+
+        if name == "facade":
+            self.new_status = CollectionState.UPDATE.value
+
+
+    #Get repo urls based on passed in info.
+    def get_valid_repos(self,session):
+        #getattr(CollectionStatus,f"{hook}_status" ) represents the status of the given hook
+        #Get the count of repos that are currently running this collection hook
+        #status_column = f"{hook}_status"
+        active_repo_count = len(session.query(CollectionStatus).filter(getattr(CollectionStatus,f"{self.name}_status" ) == CollectionState.COLLECTING.value).all())
+
+        #Will always disallow errored repos and repos that are already collecting
+
+        #The maximum amount of repos to schedule is affected by the existing repos running tasks
+        limit = self.max_repo-active_repo_count
+
+
+        user_list = split_random_users_list(session,f"{self.name}_status",self.new_status)
+
+        #Extract the user id from the randomized list and split into four chunks
+        split_user_list = split_list_into_chunks([row[0] for row in user_list], 4)
+
+        session.logger.info(f"User_list: {split_user_list}")
+
+        #Iterate through each fourth of the users fetched
+        for quarter_list in split_user_list:
+            if limit <= 0:
+                return
+
+            collection_list = get_valid_repos_for_users(session,limit,tuple(quarter_list),hook=self.name, days_to_wait_until_next_collection=self.days_until_collect_again)
+
+            self.repo_list.extend(collection_list)
+            #Update limit with amount of repos started
+            limit -= len(collection_list)
+
+        #Now start old repos if there is space to do so.
+        if limit <= 0:
+            return
+
+        #Get a list of all users.
+        query = s.sql.text("""
+            SELECT  
+            user_id
+            FROM augur_operations.users
+        """)
```

Comment:
Same thing with this as I mentioned above. This could be refactored out into get_users()